### PR TITLE
Feature/mrtk sharing2

### PR DESCRIPTION
**Pull request by mr0ng**
Overview
---
Draft architecture documents added for PR review. 

Changes
---
- Fixes: # .

### CODE REVIEW

File: Assets/OngSharing/ClickSharer.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.SDK.Input.Handlers;
+//using Microsoft.MixedReality.Toolkit.InputSystem;
+//using HoloToolkit.Unity.InputModule;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ClickSharer : BaseFocusHandler, IMixedRealityInputHandler, IMixedRealityPointerHandler, IMixedRealityFocusChangedHandler
```
Comment (StephenHodgson) [inline Assets/OngSharing/ClickSharer.cs]: For now, let's try to keep it just to documentation.

File: Documentation/Architecture/INetworkingBase.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Properties
+
+### NetworkingMode
+
+This is the selected type of networking that will be used.
+It is selected from the following NetworkingType Enum: [NetworkingType](./NetworkingType.md)
+
+Is set to Photon by default.
+
```
Comment (jessemcculloch) [inline Documentation/Architecture/INetworkingBase.md]: This should not be in the base, but rather provided by the concrete class

File: Documentation/Architecture/INetworkingBase.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+| Type |
+| --- |
+| int |
+
+### defaultReliability
+
+| Type |
+| --- |
+| int |
```
Comment (jessemcculloch) [inline Documentation/Architecture/INetworkingBase.md]: Float

File: Documentation/Architecture/INetworkingBase.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+### Send(int ID, T data)
+
+
+### Send(T data, float reliability)
+
+
+### Send(int ID, T data, float reliability)
+
+
+
```
Comment (jessemcculloch) [inline Documentation/Architecture/INetworkingBase.md]: Need a data field to specify data type (video, audio, etc)

File: Documentation/Architecture/INetworkingBase.md (hunk 2 above)
Comment (StephenHodgson) [reply Documentation/Architecture/INetworkingBase.md]: `IMixedRealityNetworkSystem`

Comment (StephenHodgson) [reply Documentation/Architecture/INetworkingBase.md]: Also, why are we going to set it to photon by default?

Shouldn't we have it set to either uNET (as it's built into the editor) or our planned WebRTC default?

File: ProjectSettings/UnityConnectSettings.asset [hunk 5]
```diff
@@ -3,7 +3,7 @@
 --- !u!310 &1
 UnityConnectSettings:
   m_ObjectHideFlags: 0
-  m_Enabled: 0
+  m_Enabled: 1
```
Comment (StephenHodgson) [inline ProjectSettings/UnityConnectSettings.asset]: please revert

File: ProjectSettings/ProjectSettings.asset [hunk 6]
```diff
@@ -627,7 +627,7 @@ PlayerSettings:
   metroCertificatePassword: 
   metroCertificateSubject: Microsoft
   metroCertificateIssuer: Microsoft
-  metroCertificateNotAfter: 0090761b2802d501
+  metroCertificateNotAfter: 00eb1f507373d501
```
Comment (StephenHodgson) [inline ProjectSettings/ProjectSettings.asset]: please revert

File: .gitignore [hunk 7]
```diff
@@ -54,3 +54,4 @@ Assets/ThirdParty/
 Assets/ThirdParty.meta
 Assets/TextMesh Pro.meta
 Assets/TextMesh Pro/
+MRTK vNEXT with Ong Sharing and Photon.zip
```
Comment (StephenHodgson) [inline .gitignore]: Where is this file?

File: Assets/OngSharing/IPhotonNetworking.cs [hunk 8]
```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public class IPhotonNetworking<T> : Singleton<IPhotonNetworking<T>> {
```
Comment (StephenHodgson) [inline Assets/OngSharing/IPhotonNetworking.cs]: We shouldn't be using Singleton. Please update this to use a `IMixedRealityNetworkingSystem` that inherits from `IMixedRealityService`. Then register it in the "Additional Service Providers Profile"

File: Assets/OngSharing/IPhotonNetworking.cs [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+            //if (NED == null)
+            //{
+            //    NED = new NetworkEventData<T>(EventSystem.current);
+            //}
+            //NED.value = dataStream;
+
+            NED.Initialize(dataStream);
+
+            getObjectById(Identity).SendMessage("OnDataReceived", NED);
```
Comment (StephenHodgson) [inline Assets/OngSharing/IPhotonNetworking.cs]: Whoa? Send Message?

File: Assets/OngSharing/NetworkingType.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
+﻿public enum NetworkingType
+{
+    /// <summary>
+    /// A very common networking method for unity which has been used as the default networking type.
+    /// </summary>
+    Photon,
```
Comment (StephenHodgson) [inline Assets/OngSharing/NetworkingType.cs]: Let's have a `None = 0,` initializer

**Comment by SimonDarksideJ** [comment]
@mr0ng can we get an update on this PR?  Feature branch PR's should be fairly quick to merge.

Also make sure you are ONLY committing files related to your change and not Unity project assets! :D

**Comment by StephenHodgson** [comment]
Closing this PR into the sharing branch as we've created a preview branch specifically for PUN.

Initial PRs for sharing should be arch docs and interfaces.

**Pull request closed without merging**